ui/pages/LoginPage.qml

```diff
@@ -34,17 +46,23 @@ SimpleLayout {
             Layout.alignment: Qt.AlignHCenter
 
             text: qsTr("SSO")
-
-            onPressed:
+            visible: !Auth.configured
+            onClicked:
             {
-             Store.register("OAuth2")
+                Store.register("OAuth2")
```

Comment:
This is not necessary to use Store.register but if you need it anyway, you can use Component.onCompleted to load it and avoid 2 buttons.
Loading component can be used to show a spinner during configuration loading (!Auth.configured) too as another possibility

Reply:
I tried the Component.onCompleted event and I get an error because of the web socket not opened : 
qml: socker error: Messages can only be sent when the socket is open.

We will go with two button for now and I will try to get an event on the websocket open to do the Store.register

Reply:
In fact the auth config should come from a REST API and not the websocket itself. This is what was discussed for SSO but the implementation didn't follow it.
Once authenticated, the connection to the websocket should be done with the JWT to secure it.

But I understand you point for the temporary solution.